sys/arm64/arm64/locore.S

```diff
@@ -880,6 +916,12 @@ ENTRY(sigcode)
 	svc	0
 
 	b	1b
+#if __has_feature(capabilities)
+#ifndef __CHERI_PURE_CAPABILITY__
+	.arch_extension noc64
+	.arch_extension a64c
```

Comment:
We should probably make this and the `.arch_extension` above macros, similar to `ENTER_C64`/`EXIT_C64`, but without the `bx #4` instruction.

Reply:
They're only used in one place, and will rarely need to be used, unlike ENTER_C64/EXIT_C64, and giving them precise-enough names that it's not confusing will be rather unwieldy (since you need to think very carefully about all three of the nocheri/hybrid/purecap kernel cases).

Reply:
For RISC-V we just used the bare assembly annotations to switch modes (`.option capmode` and `.option nocapmode`).  RISC-V is a bit nicer here as we can use `.option push` and `.option pop` to reduce the number of purecap vs hybrid #ifdef's needed.  The purecap kernel will need another set of these around `freebsd64_sigcode`, but they will be the opposite, so I think in this instance it will be clearer to just leave them as-is.

Reply:
(We also didn't have to do mode switching for copy* or f/su* in RISC-V for hybrid, so the annotations were only used in a couple of places in locore and not as widely.  I think ENTER_C64/EXIT_C64 are useful given they are needed for copy* or f/su*, I'm just not sure it's as useful for these cases which would be a macro defined and then used only in a single place.)